### PR TITLE
TransformControls: Fix edge case transforms.

### LINKED ISSUE #21840: Using TransformControls to move an object along the axis, the path is not correct
**Describe the bug**

Using TransformControls to move an object along the axis, the path is not correct

**To Reproduce**

Use TransformControls to move the object along the axis. When the following three conditions are met, the path is not correct

1. This object has a parent object that has been rotated

2. The coordinate space is 'world'

3. The translationSnap is a a numeric value

***Code***
TransformControls.js  line 421
```
if ( space === 'world' ) {
  // if ( object.parent ) {
  // object.position.add( _tempVector.setFromMatrixPosition( object.parent.matrixWorld ) );
  // }
  // if ( axis.search( 'X' ) !== - 1 ) {
  // 	object.position.x = Math.round( object.position.x / this.translationSnap ) * this.translationSnap;
  // }
  // if ( axis.search( 'Y' ) !== - 1 ) {
  // 	object.position.y = Math.round( object.position.y / this.translationSnap ) * this.translationSnap;
  // }
  // if ( axis.search( 'Z' ) !== - 1 ) {
  // 	object.position.z = Math.round( object.position.z / this.translationSnap ) * this.translationSnap;
  // }
  // if ( object.parent ) {
  // 	object.position.sub( _tempVector.setFromMatrixPosition( object.parent.matrixWorld ) );
  // }
  var vector_temp = new Vector3();
  object.getWorldPosition(vector_temp);
  if ( axis.search( 'X' ) !== - 1 ) {
    vector_temp.x = Math.round( vector_temp.x / this.translationSnap ) * this.translationSnap;
  }
  if ( axis.search( 'Y' ) !== - 1 ) {
    vector_temp.y = Math.round( vector_temp.y / this.translationSnap ) * this.translationSnap;
  }
  if ( axis.search( 'Z' ) !== - 1 ) {
    vector_temp.z = Math.round( vector_temp.z / this.translationSnap ) * this.translationSnap;
  }
  object.position.copy(object.parent.worldToLocal(vector_temp))
}
```

***Live example***

**Expected behavior**

Using TransformControls to move an object along the axis, the path is correct.

**Screenshots**

5f5b9b60161e38e6d8e23cadc066b62
40fbe4deb93aa81fe1da913d23ebedf

**Platform:**

 - Device: [Desktop]
 - OS: [Windows]
 - Browser: [Chrome]
 - Three.js version: [dev, r123]

### PR DESCRIPTION
**Pull request by AlaricBaraou**
Fixed #21840

**Description**

Unexpected transform happens when using TransformControls under theese three specific conditions:

-  When the object has a rotated parent
- The coordinate space is set to 'world'
- TranslationSnap is a numeric value

Addiotionnaly if the scene had a rotation, it would also mess up the TransformControls expected behaviour.

Both have been fixed in this PR.

*note, changed a snap value in the demo that was wayy to high to be usable with translate

https://github.com/mrdoob/three.js/assets/7174039/44ba9f08-f25a-4b81-b78d-f90a83573b5e

* closed the previous PR https://github.com/mrdoob/three.js/pull/27718 after forgetting to remove a line

*This contribution is funded by [Alaric Baraou 🦦](https://alaricbaraou.com/)*

### CODE REVIEW

File: examples/jsm/controls/TransformControls.js [hunk 1]
```diff
@@ -215,7 +215,13 @@ class TransformControls extends Object3D {
...
 		}
 
-		super.updateMatrixWorld( force );
+		// TransformControls should ignore any parent matrix
+		this.matrixAutoUpdate = false;
+		this.updateMatrix();
+		this.matrixWorldNeedsUpdate = false;
+		this.matrixWorld.copy( this.matrix );
+
+		super.updateMatrixWorld( false );
```
Comment (Mugen87) [inline examples/jsm/controls/TransformControls.js:224]: I can confirm your PR fixes the issue but the approach isn't ideal, tbh. We need to find a solution _without_ influencing how the world transformation matrix is computed. 

If not, this will cause issues when we rethink the update matrix process and lifecycle in `three.js` and try to find a more optimized implementation. Ideally, components just request a world matrix and do not modify their computation.

In other words, `TransformControls` has to work with an unmodified version of the scene's hierarchy. Fixing the controls in combination with rotated ancestor nodes has to be done in a way that the rotation is correctly taken into account and not ignored.

Comment (AlaricBaraou) [reply examples/jsm/controls/TransformControls.js:224]: Would this match your requirements ? 

```
// if TransformControls has a parent, set its local position to the inverse of the parent in order to have a worldMatrix to identity
if ( this.parent ) {

			this._transformMatrixWorldInv.copy( this.parent.matrixWorld ).invert();
			this._transformMatrixWorldInv.decompose( this.position, this.rotation, this.scale );

}

super.updateMatrixWorld( true );
```

That way we don't influence how the world transformation matrix is computed, we just set the TransformControls local position.

Comment (Mugen87) [reply examples/jsm/controls/TransformControls.js:224]: That looks indeed better.

Comment (AlaricBaraou) [reply examples/jsm/controls/TransformControls.js:224]: Awesome! 
I'll push that commit later today

Comment (AlaricBaraou) [reply examples/jsm/controls/TransformControls.js]: updated, it should be good now 👍
